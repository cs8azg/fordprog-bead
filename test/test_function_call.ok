function natural five()
begin
  return 5
end

function natural add(natural a, natural b)
  natural sum
begin
  sum := a + b
  return sum
end

function natural minus(natural a, natural b)
  natural diff
begin
  if b > a then
    diff := 42
  else
    diff := (a - b)
  endif
  return diff
end

function boolean is_between_5_and_10(natural a)
begin
  if a < 5 then
    return false
  endif
  return a <= 10
end

function boolean to_boolean(natural number)
begin
  if number > 0 then
    return true
  else
    return false
  endif
end

function natural ten()
begin
  return add(4, 6)
end

program test_function_call
  natural a
  natural b
  natural sum
begin
  write(five())
  a := 2
  b := 7
  sum := add (a, b)
  write(sum)
  write(add(a, 5))
  write(add(2, 4))
  write(add(add(a, b), b))
  write(minus(b, a))
  write(minus(a, b))
  write(is_between_5_and_10(3))
  write(is_between_5_and_10(7))
  write(is_between_5_and_10(12))
  write(to_boolean(5))
  write(to_boolean(0))
  write(ten())
end
