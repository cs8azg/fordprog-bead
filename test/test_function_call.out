5
9
7
6
16
5
false
true
false